cmd/populator/populator.go

```diff
@@ -45,31 +46,30 @@ func main() {
 	opts.BindFlags(goflag.CommandLine)
 	flag.Parse()
 
-	if err := populate(zap.New(zap.UseFlagOptions(&opts)), image, devicePath); err != nil {
+	log := zap.New(zap.UseFlagOptions(&opts))
+	if err := populate(log, image); err != nil {
+		log.Error(err, "Image population failed")
 		os.Exit(1)
 	}
 }
 
-func populate(log logr.Logger, ref string, devicePath string) error {
+func populate(log logr.Logger, ref string) error {
 	ctx := ctrl.SetupSignalHandler()
 	log.Info("Starting image population")
 
 	reg, err := remote.DockerRegistry(nil)
 	if err != nil {
-		log.Error(err, "Failed to initialize registry")
-		return err
+		return fmt.Errorf("failed to initialize registry")
```

Comment:
You want to wrap the error here via `fmt.Errorf("....: %w", err)`

Reply:
Same applies for the other error returns

Reply:
copy&paste 🤦‍♂️